src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+#include "global.h"
+#include "intro_movie_internal.h"
+#include "math_util.h"
+#include "system.h"
+#include "sys_task_api.h"
+#include "unk_0200FA24.h"
+
+void Task_IntroMovie_BlendFadeEffect(SysTask *task, void *pVoid);
+int IntroMovie_BgLayerToScrollEffectSlot(enum GFBgLayer bgId);
+void Task_IntroMovie_BgScroll_VBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_BgScroll_NotVBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_WindowPanEffect(SysTask *task, void *pVoid);
+void Task_IntroMovie_CircleWipeEffect(SysTask *task, void *pVoid);
+void IntroMovie_CircleWipeEffect_HBlankCB(void *pVoid);
+
+void IntroMovie_CreateSpriteResourceManagers(IntroMovieOvyData *data, const u8 *counts) {
+    for (u8 i = 0; i < 4; ++i) {
+        data->spriteResManagers[i] = Create2DGfxResObjMan(counts[i], (GfGfxResType)i, HEAP_ID_INTRO_MOVIE);
+    }
+}
+
+void IntroMovie_DestroySpriteResourceManagers(IntroMovieOvyData *data) {
+    for (u8 i = 0; i < 4; ++i) {
+        Destroy2DGfxResObjMan(data->spriteResManagers[i]);
+    }
+}
+
+GF_2DGfxResMan **IntroMovie_GetSpriteResourceManagersArray(IntroMovieOvyData *data) {
+    return data->spriteResManagers;
+}
+
+void IntroMovie_StartSpriteAnimAndMakeVisible(Sprite *sprite, BOOL active) {
+    Set2dSpriteAnimActiveFlag(sprite, active);
+    Set2dSpriteVisibleFlag(sprite, active);
+}
+
+void IntroMovie_BuildSpriteResourcesHeaderAndTemplate(int resId, IntroMovieOvyData *data, int priority, NNS_G2D_VRAM_TYPE whichScreen, SpriteTemplate *template, SpriteResourcesHeader *header) {
+    CreateSpriteResourcesHeader(
+        header,
```

Comment:
ew, pls dont format it like this, if you have to split it over multiple lines, don't do it for every param, just once or twice when it gets too long

Reply:
vehemently disagree

Reply:
style has agreed with this

Reply:
make the .clang-format pr and we'll talk. for now, the code is fine (and infinitely more readable) as is

Reply:
strong disagree, but you do have a point about the PR, that is something that I really need to do